.travis.yml

```diff
@@ -1,22 +1,24 @@
+sudo: false
 language: emacs-lisp
 env:
-  - EMACS=emacs24
-  - EMACS=emacs-snapshot
+  - EMACS_BINARY=emacs-24.4-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-24.4-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-24.5-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-24.5-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.1-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-25.1-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test-bytecomp
+  - EMACS_BINARY=emacs-25.2-travis MAKE_TEST=test-checks
+  - EMACS_BINARY=emacs-git-snapshot-travis
```

Comment:
Is such a big matrix really needed? How about saving some trees and unifying all tests per emacs version?